t/t7500-commit.sh

```diff
@@ -359,4 +359,36 @@ test_expect_success 'new line found before status message in commit template' '
 	test_i18ncmp expected-template editor-input
 '
 
+# Tests for commit.allowEmpty config
+
+test_expect_success "no commit.allowEmpty and no --allow-empty" "
+	test_must_fail git commit -m 'test'
+"
+
+test_expect_success "no commit.allowEmpty and --allow-empty" "
+	git commit --allow-empty -m 'test'
+"
+
+for i in true 1
+do
+	test_expect_success "commit.allowEmpty=$i and no --allow-empty" "
+		git -c commit.allowEmpty=$i commit -m 'test'
+	"
+
+	test_expect_success "commit.allowEmpty=$i and --allow-empty" "
+		git -c commit.allowEmpty=$i commit --allow-empty -m 'test'
+	"
+done
+
+for i in false 0
+do
+	test_expect_success "commit.allowEmpty=$i and no --allow-empty" "
+		test_must_fail git -C commit.allowEmpty=$i commit -m 'test'
+	"
+
+	test_expect_success "commit.allowEmpty=$i and --allow-empty" "
+		test_must_fail git -c commit.allowEmpty=$i commit --allow-empty -m 'test'
+	"
+done
+
```

Comment:
Wow, you went all out ;-)

However, for such a minor feature I don't think we should add that much time to run the test suite. Maybe a very simple one would do instead? Something like:

```sh
test_expect_success 'commit.allowEmpty' '
    test_tick &&
    git -c commit.allowEmpty=true commit -m empty
'
```

Reply:
hey, 
I submitted this patch/PR at git list git@vger.kernel.org, and they pointed out that the tests doesn't address the issue https://github.com/git-for-windows/git/issues/1854.
So, I replaced them with this one:
```
test_expect_success 'cherry pick existing commit with commit.allowEmpty'  '
    test_tick &&
	test_commit "first" &&
	test_commit "second" &&
	git -c commit.allowEmpty=true cherry-pick HEAD~1
'
```